mixins/rtl-mixin.js

```diff
@@ -13,9 +14,25 @@ export const RtlMixin = dedupeMixin(superclass => class extends superclass {
 
 	constructor() {
 		super();
+		this._localeSettings = getDocumentLocaleSettings();
+		this._handleLanguageChange = this._handleLanguageChange.bind(this);
+		this._handleLanguageChange();
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		this._localeSettings.addChangeListener(this._handleLanguageChange);
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		this._localeSettings.removeChangeListener(this._handleLanguageChange);
+	}
+
+	_handleLanguageChange() {
 		const dir = document.documentElement.getAttribute('dir');
 		// avoid reflecting "ltr" for better performance
-		if (dir && dir !== 'ltr') {
+		if (dir && (dir !== 'ltr' || this.dir === 'rtl')) {
```

Comment:
This logic needed slight tweaking to handle a change after the initial value was set. If things go from `rtl` -> `ltr`, we DO need to reflect that change. I could also set it back to `undefined` -- what do folks think?

Reply:
Is it valid to assign `this.dir = undefined`? When I log `dir` for elements without an explicit `dir` assignment, the browser returns an empty string, though [Mozilla says](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dir) the property's values can be `ltr`, `rtl`, and `auto`.

Reply:
Hmm, interesting. I just checked, and even before these changes, any component using `RtlMixin` was having its `dir` set to `undefined` by default. I'm guessing Lit's doing that since we're redefining `dir` as a property and aren't initializing it in the constructor.

If we're worried about it being `undefined`, we could _not_ define it as a Lit property (since it's already a native one) and just manually call `requestUpdate` when we set it.

Reply:
Cool - I'm happy either way as long as the browser doesn't 🤮 .